From beab3f886e8b314848290f172ec4c526bdaeb434 Mon Sep 17 00:00:00 2001
From: Simone Schwizer <simone.schwizer@bytesatwork.ch>
Date: Wed, 13 Aug 2025 15:20:30 +0200
Subject: [PATCH] bytesatwork: bawconfig: Add imx8mp HW revision and RAM type

Add PCB revision 1.1 and 2GB RAM from ISSI.

Signed-off-by: Simone Schwizer <simone.schwizer@bytesatwork.ch>
(cherry picked from commit 032f569ed0e241efaf2428a26ce111e17d4c34d6)
---
 board/bytesatwork/baw_config/baw_config.c | 2 ++
 board/bytesatwork/baw_config/baw_config.h | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/board/bytesatwork/baw_config/baw_config.c b/board/bytesatwork/baw_config/baw_config.c
index 87024b01eb..2a63fe3810 100644
--- a/board/bytesatwork/baw_config/baw_config.c
+++ b/board/bytesatwork/baw_config/baw_config.c
@@ -23,6 +23,7 @@ const struct baw_config_name_pair baw_config_pcb_name[] = {
 	{ M8P_PCB_REV_1_1, "M8P Rev 1.1" },
 	{ M9_PCB_REV_1_0, "M9 Rev 1.0" },
 	{ M9_PCB_REV_2_0, "M9 Rev 2.0" },
+	{ M7_PCB_REV_1_1, "M7 Rev 1.1" },
 	{ -1, NULL },
 };
 
@@ -56,6 +57,7 @@ const struct baw_config_name_pair baw_config_ram_name[] = {
 	{ M9_RAM_MT53D512M16D1DS_046, "MT53D512M16D1DS-046 1 GB LPDDR4X @ 1866 MHz" },
 	{ M9_RAM_MT53E1G16D1FW_046, "MT53E1G16D1FW-046 2 GB LPDDR4X @ 1866 MHz" },
 	{ M7_RAM_MT53E1536M32D4DE_046, "MT53E1536M32D4DE-046 6 GB LPDDR4 @ 2000 MHz"},
+	{ M7_RAM_IS43LQ32512A_046, "IS43LQ32512A-046BLI 2GB LPDDR4 @ 2000 MHz"},
 	{ -1, NULL },
 };
 
diff --git a/board/bytesatwork/baw_config/baw_config.h b/board/bytesatwork/baw_config/baw_config.h
index f1f3262386..8a86b4a9c2 100644
--- a/board/bytesatwork/baw_config/baw_config.h
+++ b/board/bytesatwork/baw_config/baw_config.h
@@ -27,6 +27,7 @@ enum baw_config_pcb {
 	M8P_PCB_REV_1_1 = 14,
 	M9_PCB_REV_1_0 = 15,
 	M9_PCB_REV_2_0 = 16,
+	M7_PCB_REV_1_1 = 17,
 };
 
 enum baw_config_ram {
@@ -62,6 +63,7 @@ enum baw_config_ram {
 	M9_RAM_MT53D512M16D1DS_046 = 27,
 	M9_RAM_MT53E1G16D1FW_046 = 28,
 	M7_RAM_MT53E1536M32D4DE_046 = 29,
+	M7_RAM_IS43LQ32512A_046 = 30,
 };
 
 enum baw_config_flash {
